lib/workload/stateful/filemanager/database/migrations/0001_add_object_table.sql

```diff
@@ -7,13 +7,13 @@ create table object (
     -- The name of the object.
     key varchar(1024) not null,
     -- The size of the object.
-    size int not null,
+    size int default null,
     -- A unique identifier for the object, if it is present.
     hash varchar(255) default null,
```

Comment:
I know it isn't in this PR - but I am a bit suspicious about this modelling.
I think you'll end up in scenarios with multiple hashes (possibly needing to know this difference between them - so you can pass on a SHA-256 hash downstream say)

Reply:
Also, this is a unique identifier for the "content" of the object..

Reply:
Yes... I'm imagining some scenario where someone wants a SHA-256 rather than just a check to see if this object is the same? I think a simple solution is to just have multiple fields for some common checksum types, e.g. SHA-256, MD5, etc (or have a type field which indicates the checksum algorithm). This also maps nicely to the [`HeadObject`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html) outputs. Although I don't know if it's promised what kind of checksum the AWS etag is, I think it's MD5?

Reply:
etag is MD5 *until* the object gets above a certain size (which is definite for bioinformatics) - at which point it is not an MD5. Best to define it as its own special type "AWS-ETAG"

Reply:
Hm, sounds like we'll have to tackle some sort of computationally lightweight hashing scheme (via some reliably unique subsampling perhaps?)... totally part of another PR, agree.